src/lib/data/configuration/options.py

```diff
@@ -0,0 +1,332 @@
+from typing import Any
+from typing import List
+from typing import Optional
+
+from src.utils.data_control import from_list
+from src.utils.data_control import from_none
+from src.utils.data_control import from_str
+from src.utils.data_control import from_stringified_bool
+from src.utils.data_control import from_union
+from src.utils.data_control import is_type
+
+
+class Options:
+    exclude_tags: Optional[List[str]]
+    source_mode: bool
+    user_rules_directory: Optional[List[str]]
+    disable_tattletale: Optional[bool]
+    additional_classpath: Optional[List[str]]
+    include_tags: Optional[List[str]]
+    source: Optional[List[str]]
+    enable_transaction_analysis: Optional[bool]
+    packages: Optional[List[str]]
+    enable_compatible_files_report: Optional[bool]
+    target: List[str]
+    output: str
+    enable_class_not_found_analysis: Optional[bool]
+    input: Optional[List[str]]
+    user_ignore_path: Optional[List[str]]
+    exclude_packages: Optional[List[str]]
+    mavenize: Optional[bool]
+    exploded_app: Optional[bool]
+    keep_work_dirs: Optional[bool]
+    online: Optional[bool]
+    windup_cli: str
+    skip_reports: Optional[bool]
+    overwrite: Optional[bool]
+    export_csv: Optional[bool]
```

Comment:
Let's organize these with mandatory items on top and optional ones at the bottom as it is the syntax in coding languages .

Reply:
done